### PR TITLE
feat: add device fields

### PR DESCRIPTION
**Pull request by peterdeme**
Add new device fields

### CODE REVIEW

File: lib/GetStream/StreamChat/Client.php [hunk 1]
```diff
@@ -870,12 +870,13 @@ public function deleteCommand(string $name): StreamResponse
...
      * @throws StreamException
      */
-    public function addDevice(string $deviceId, string $pushProvider, string $userId): StreamResponse
+    public function addDevice(string $deviceId, string $pushProvider, string $userId, string $pushProviderName = null): StreamResponse
     {
         $data = [
             "id" => $deviceId,
             "push_provider" => $pushProvider,
             "user_id" => $userId,
+            "push_provider_name" => $pushProviderName
```
[the remark this answers is not in the document]
Comment (ferhatelmas) [reply lib/GetStream/StreamChat/Client.php:879]: don't we have check_push support here either?

[the remark this answers is not in the document]
Comment (peterdeme) [reply lib/GetStream/StreamChat/Client.php:879]: I think we should refactor this to:
```
addDevice($device array)
```

[the remark this answers is not in the document]
Comment (ferhatelmas) [reply lib/GetStream/StreamChat/Client.php:879]: we could support both

[the remark this answers is not in the document]
Comment (peterdeme) [reply lib/GetStream/StreamChat/Client.php]: for now I just reordered this array.

as for checkpush, we don't need to do anything there: https://github.com/GetStream/stream-chat-php/blob/d39c770e10df1a96568c7f126740c2acd33f78bc/lib/GetStream/StreamChat/Client.php#L287

[the remark this answers is not in the document]
Comment (ferhatelmas) [reply lib/GetStream/StreamChat/Client.php]: both sounds good

**Review by ferhatelmas — APPROVED** [review]

**Pull request merged**